File: app/src/main/java/com/main/climbingdiary/dialog/DialogFactory.java
package com.main.climbingdiary.dialog;

import com.main.climbingdiary.activities.MainActivity;
import com.main.climbingdiary.database.entities.Route;
import com.main.climbingdiary.models.Tabs;

public class DialogFactory {

    public static void openAddRouteDialog(Tabs tab) {
        if (tab == Tabs.PROJEKTE) {
            AddProjektDialog addProjekt = new AddProjektDialog("Neues Projekt");
            addProjekt.show(MainActivity.getMainActivity().getSupportFragmentManager(), "fragment_add_Projekt");
        } else {
            AddRouteDialog addRoute = new AddRouteDialog("Neue Route");
            addRoute.show(MainActivity.getMainActivity().getSupportFragmentManager(), "fragment_add_Route");
        }
    }

    public static void openEditRouteDialog(Tabs tab, int _id) {
        if (tab == Tabs.PROJEKTE) {
            EditProjektDialog editDialog = new EditProjektDialog("Projekt bearbeiten", _id);
            editDialog.show(MainActivity.getMainActivity().getSupportFragmentManager(), "edit");
        } else {
            EditRouteDialog editDialog = new EditRouteDialog("Route bearbeiten", _id);
            editDialog.show(MainActivity.getMainActivity().getSupportFragmentManager(), "edit");
        }
    }

    public static void openEditRouteDialog(Route _route) {
        EditRouteDialog editDialog = new EditRouteDialog("Route bearbeiten", _route);
        editDialog.show(MainActivity.getMainActivity().getSupportFragmentManager(), "edit");
    }
}

File: app/src/main/java/com/main/climbingdiary/dialog/DialogFactory.kt
package com.main.climbingdiary.dialog

import androidx.fragment.app.FragmentManager
import com.main.climbingdiary.activities.MainActivity
import com.main.climbingdiary.database.entities.Route
import com.main.climbingdiary.models.Tabs

object DialogFactory {

    private val manager: FragmentManager by lazy { MainActivity.getMainActivity().supportFragmentManager }

    fun openAddRouteDialog(tab: Tabs) {
        if (tab === Tabs.PROJEKTE) {
            val addProjekt = AddProjektDialog("Neues Projekt")
            addProjekt.show(
                manager,
                "fragment_add_Projekt"
            )
        } else {
            val addRoute = AddRouteDialog("Neue Route")
            addRoute.show(
                manager,
                "fragment_add_Route"
            )
        }
    }
    fun openEditRouteDialog(tab: Tabs, id: Int) {
        if (tab === Tabs.PROJEKTE) {
            val editDialog = EditProjektDialog("Projekt bearbeiten", id)
            editDialog.show(manager, "edit")
        } else {
            val editDialog = EditRouteDialog("Route bearbeiten", id)
            editDialog.show(manager, "edit")
        }
    }

    fun openTickProjektDialog(projektId: Int) {
        val editDialog = EditRouteDialog("Projekt ticken", projektId,true)
        editDialog.show(manager, "tick")
    }
}